### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'mutations')

### ISSUE THREAD

**Issue by jamesdaniels** [firebase/firebase-js-sdk#7721]
### Operating System

Various

### Browser Version

Chrome/118.0.5993.88

### Firebase SDK Version

10.5.0

### Firebase SDK Product:

Firestore

### Describe your project's tooling

Angular 16.2.7, Karma 6.4.2, Node 18

### Describe the problem

In my [karma tests (against the emulators)](https://github.com/angular/angularfire/actions/runs/6635408692/job/18026289802) I'm running into tons of failures with Firestore, one culprit appears to be this `!`:
https://github.com/firebase/firebase-js-sdk/blob/f002ef36a6b427fd526696f9cd6077a217ccc6ef/packages/firestore/src/remote/remote_store.ts#L779

Stack trace:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'mutations')
    at Function.from (index.esm2017.js:5483:32)
    at index.esm2017.js:14876:53
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:22:1)
    at asyncToGenerator.js:27:1
    at ZoneAwarePromise (zone.js:1411:21)
    at asyncToGenerator.js:19:1
    at __PRIVATE_onMutationResult (index.esm2017.js:14874:42)
    at __PRIVATE_PersistentWriteStream.onMessage (index.esm2017.js:14334:3
```

Seems to be related to a bunch of these too, maybe failed/aborted tests put Firestore into a bad state?

```
Uncaught Error: FIRESTORE (10.5.0) INTERNAL ASSERTION FAILED: Unexpected state
    at fail (index.esm2017.js:194:34)
    at __PRIVATE_AsyncQueueImpl.ou (index.esm2017.js:18396:20)
    at __PRIVATE_AsyncQueueImpl.enqueue (index.esm2017.js:18312:18)
    at __PRIVATE_AsyncQueueImpl.enqueueAndForget (index.esm2017.js:18297:14)
    at DelayedOperation.handleDelayElapsed (index.esm2017.js:15076:25)
    at index.esm2017.js:15058:51
    at timer (zone.js:2367:41)
    at _ZoneDelegate.invokeTask (zone.js:402:31)
    at ProxyZoneSpec.onInvokeTask (zone-testing.js:304:39)
    at _ZoneDelegate.invokeTask (zone.js:401:60)
```

### Steps and code to reproduce issue

Use Firestore in Karma (Angular's test tooling)

**Comment by jamesdaniels** [comment]
Related? https://github.com/angular/angularfire/issues/3448

**Comment by hsubox76** [comment]
I don't think it's related to that bug, that user opened what I think is the same issue in our repo: https://github.com/firebase/firebase-js-sdk/issues/7712 where the same test works if they use karma - in that case it's a module resolution issue.

About this issue, since it's using emulators I wonder if it's related to recent changes Firestore made that require the latest versions of the emulators and an EXPERIMENTAL_MODE flag? https://github.com/firebase/firebase-js-sdk/pull/7637 These don't look like the same failure though.

**Comment by tom-andersen** [comment]
@jamesdaniels We had a problem with read stream that might be affecting the write stream as well. Unfortunately, this is an intermittent problem that was only reproducible under certain conditions.

Could you upgrade to version 1.5.2 of Firebase SDK?

Also, if you have a test that consistently fails, but it fixed with 1.5.2, that would be very helpful information for us.
